pkg/mappings.bzl

```diff
@@ -249,13 +270,15 @@ pkg_files = rule(
             mandatory = True,
             allow_files = True,
         ),
-        "attributes": attr.string_list_dict(
+        "attributes": attr.string(
             doc = """Attributes to set on packaged files.
 
+            Always use `pkg_attributes()` to set this rule attribute.
+
             Consult the "Mapping Attributes" documentation in the rules_pkg
             reference for more details.
             """,
-            default = {"unix": ["-", "-", "-"]},
+            default = pkg_attributes(),
```

Comment:
This is subtly different from -,-,- given that mode defaults to 0444. 
If you want the default to be "let the builder pick" then this should be default="" or pkg_attributes should treat mode like owner and group.